File: iOS8_Advanced/5_2_PDFViewer/PDFViewer/PDFView.m
//
//  PDFView.m
//  BookViewer
//
//  Created by Masayuki Nii on 11/11/24.
//  Copyright (c) 2011年 __MyCompanyName__. All rights reserved.
//

#import "PDFView.h"

@implementation PDFView

- (id)initWithFrame: (CGRect)frame
{
#ifdef DEBUG
    NSLog( @"%s", __FUNCTION__);
#endif
    self = [super initWithFrame:frame];
    if (self) {
        // Initialization code
    }
    return self;
}

- (void)drawRect: (CGRect)rect
{
#ifdef DEBUG
    NSLog( @"%s", __FUNCTION__);
#endif
    if ( self.pageRef != nil ) {
        CGContextRef context = UIGraphicsGetCurrentContext();
        CGContextSaveGState(context);
        
 //       CGContextSetInterpolationQuality(context, kCGInterpolationHigh);
//        CGInterpolationQuality x = CGContextGetInterpolationQuality (context);
//        NSLog(@"CGInterpolationQuality=%d", x);

        
        CGContextSetFillColorWithColor( context, [UIColor whiteColor].CGColor );
        UIRectFill( rect );
        CGRect pageRect = CGPDFPageGetBoxRect( self.pageRef, kCGPDFMediaBox );
        
        CGContextConcatCTM (context, 
                            CGPDFPageGetDrawingTransform (self.pageRef, 
                                                          kCGPDFMediaBox, 
                                                          pageRect, 0, true ));
        CGContextConcatCTM (context, 
                            CGAffineTransformMakeTranslation( 0, pageRect.size.height) );
        CGContextConcatCTM (context, 
                            CGAffineTransformMakeScale( 1.0, -1.0 ) );
        CGContextClipToRect ( context, rect );
        
        CGContextDrawPDFPage(context, self.pageRef);
        CGContextRestoreGState(context);
    }
}

@end

File: iOS8_Advanced/6_2_PDFViewer_Swift/PDFViewer_Swift/PDFView.swift
//
//  PDFView.swift
//  PDFViewer_Swift
//
//  Created by Masayuki Nii on 2015/05/24.
//  Copyright (c) 2015年 msyk.net. All rights reserved.
//

import UIKit

class PDFView: UIView {

    var pageRef : CGPDFPageRef?

    required override init(frame: CGRect) {
        super.init(frame: frame)
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func drawRect(rect: CGRect) {
        if ( self.pageRef != nil ) {
            let context = UIGraphicsGetCurrentContext();
            CGContextSaveGState(context);
            
            //       CGContextSetInterpolationQuality(context, kCGInterpolationHigh);
            //        CGInterpolationQuality x = CGContextGetInterpolationQuality (context);
            //        NSLog(@"CGInterpolationQuality=%d", x);
            
            
            CGContextSetFillColorWithColor( context, UIColor.whiteColor().CGColor )
            UIRectFill( rect )
            let pageRect = CGPDFPageGetBoxRect( self.pageRef, kCGPDFMediaBox )
            
            CGContextConcatCTM (context,
                CGPDFPageGetDrawingTransform (self.pageRef,
                    kCGPDFMediaBox,
                    pageRect, 0, true ));
            CGContextConcatCTM (context,
                CGAffineTransformMakeTranslation( 0, pageRect.size.height) );
            CGContextConcatCTM (context,
                CGAffineTransformMakeScale( 1.0, -1.0 ) );
            CGContextClipToRect ( context, rect );
            
            CGContextDrawPDFPage(context, self.pageRef);
            CGContextRestoreGState(context);
        
        }
    }

}
